provider/provider.go

```diff
@@ -94,3 +94,17 @@ func getDefaultPath() string {
 		return "/usr/local/lib/summon"
 	}
 }
+
+//Create slice of all file names in the default path
+func GetAllProviders() []string{
+	files, err := ioutil.ReadDir(DefaultPath)
```

Comment:
Indentation of this code is pretty odd. run your new files through golint/gofmt to check if they're correct.